validator/client/service.go

```diff
@@ -283,15 +275,8 @@ func (v *ValidatorService) ProposerSettings() *validatorserviceconfig.ProposerSe
 
 // SetProposerSettings sets the proposer settings on the validator service as well as the underlying validator
 func (v *ValidatorService) SetProposerSettings(ctx context.Context, settings *validatorserviceconfig.ProposerSettings) error {
-	if v.db == nil {
-		return errors.New("db is not set")
-	}
-	if err := v.db.SaveProposerSettings(ctx, settings); err != nil {
-		return err
-	}
 	v.proposerSettings = settings
```

Comment:
It reads a bit weird that we do 
```go
v.proposerSettings = settings
return v.validator.SetProposerSettings(ctx, settings)
```
as a reader, as intuitively these do the same thing unless we add a better comment or naming to explain this